packages/@react-native-windows/codegen/src/generators/GenerateTypeScript.ts

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License.
+ * @format
+ */
+
+'use strict';
+
+import {
+  NamedShape,
+  NativeModuleBaseTypeAnnotation,
+  NativeModuleFunctionTypeAnnotation,
+  NativeModuleObjectTypeAnnotation,
+  NativeModuleParamTypeAnnotation,
+  NativeModuleReturnTypeAnnotation,
+  NativeModuleSchema,
+  Nullable,
+  SchemaType,
+} from 'react-native-tscodegen';
+
+type ObjectProp = NamedShape<Nullable<NativeModuleBaseTypeAnnotation>>;
+type FunctionParam = NamedShape<Nullable<NativeModuleParamTypeAnnotation>>;
+type FunctionDecl = NamedShape<Nullable<NativeModuleFunctionTypeAnnotation>>;
+type FilesOutput = Map<string, string>;
+
+const moduleTemplate = `
+/*
+ * This file is auto-generated from a NativeModule spec file in js.
+ *
+ * This is a TypeScript turbo module definition file.
+ */
+
+// the following import statements are not actually working today
+import {TurboModule, RootTag} from 'react-native/Libraries/TurboModule/RCTExport';
```

Comment:
Should we get these added to @types/react-native?  Whats the plan here?
I also want to push Meta on whether it would make sense to export these types from 'react-native' instead of having to use the weird paths.

Reply:
2 years ago I heard from Meta people said they will export when they think turbo module is ready to public. I think only at that moment related symbols can be added to `@types/react-native`.